api/v1alpha1/lease_types.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// LeaseSpec defines the desired state of Lease
+type LeaseSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+	BeginTime metav1.Time `json:"beginTime"`
+	EndTime   metav1.Time `json:"endTime"`
+	// Minutes is redundant when we have begin/end time but it's useful for listing
+	Minutes      int    `json:"minutes"`
+	ExporterName string `json:"exporterName"`
+	ClientName   string `json:"clientName"`
+	// Adding the UID of the exporter and client can be redundant but
+	// it is useful if devices change name, while the string name
+	// is useful for humans inspecting the lease
+	ExporterUID types.UID `json:"exporterUID"`
+	ClientUID   types.UID `json:"clientUID"`
```

Comment:
I don't think it's possible to find objects by uid via the k8s API? (and you can't rename a object either)

Reply:
You can't rename, but could be deleted/recreated?, anyway if there is no ability to get them via UID it's pointless 👍

Reply:
We now have UUIDs in several different places:

1. for every CRD, it has a k8s assigned object id.
2. for every exporter it (re)generates a UUID on startup.
3. for every device it (re)generates a UUID on session initialization (which is currently recorded in Exporter status).

Thoughts:

1. exporter fetch the kubernetes generated object id at `Register` and use that instead.
2. no longer record device UUIDs in Exporter status, since they change for every session and generally is unsuitable for filtering.

Reply:
Point 1 would solve the problem of Exporter CRDs being deleted and recreated. Since the UUID recorded in CRD (and in turn, the uuid used by client in calls) no longer matches the UUID the exporter thinks itself to be of, it would happily reject these invalid calls.

Reply:
The invalid lease would still be dangling there though, but we can use `ownerReferences` to prevent this from happening.

Reply:
1. 👍 
2. Can you still call a specific device driver without the UUID on the client side?, how do you make parent relationships?

I'd avoid spending time in "2" now, I agree "1" is a good idea.

Reply:
> 2\. Can you still call a specific device driver without the UUID on the client side?, 

You always need the driver instance UUID for making driver calls. But (as of now) clients don't make use of the information in Exporter CRD other than the purpose of filtering. Driver instance UUIDs are always fetched with `GetReport` after the connection to the exporter is made.

> how do you make parent relationships?

That's a problem, let's keep them there for now.

Reply:
Ahh, true, the GetReport :-) 

Let's revisit it on the future!, thanks @NickCao ! :-)